runtime/lua/vscode-neovim/api.lua

```diff
@@ -1,133 +1,205 @@
-end
+------- Requests -------
 
--- send command to vscode with range (line or char). [1, 1]-based.
-function M.call_range(command, line1, line2, leaveSelection, ...)
-  return M.call_extension("range-command", command, "V", line1, line2, 1, 1, leaveSelection, { ... })
-end
+local REQUEST_STATE = {
+  id = 0,
+  callbacks = {},
+}
 
-function M.notify_range(command, line1, line2, leaveSelection, ...)
-  return M.notify_extension("range-command", command, "V", line1, line2, 1, 1, leaveSelection, { ... })
+local function add_callback(callback)
+  REQUEST_STATE.id = REQUEST_STATE.id + 1
+  REQUEST_STATE.callbacks[REQUEST_STATE.id] = callback
+  return REQUEST_STATE.id
 end
 
-function M.call_range_pos(command, line1, line2, pos1, pos2, leaveSelection, ...)
-  return M.call_extension("range-command", command, "v", line1, line2, pos1, pos2, leaveSelection, { ... })
+---Invoke the callback, called by vscode
+---@param id number callback id
+---@param result any resutl
+---@param is_error boolean is error
+function M.invoke_callback(id, result, is_error)
+  vim.schedule(function()
+    local callback = REQUEST_STATE.callbacks[id]
+    REQUEST_STATE.callbacks[id] = nil
+    if callback then
+      if is_error then
+        callback(result, nil)
+      else
+        callback(nil, result)
+      end
+    end
+  end)
 end
 
-function M.notify_range_pos(command, line1, line2, pos1, pos2, leaveSelection, ...)
-  return M.notify_extension("range-command", command, "v", line1, line2, pos1, pos2, leaveSelection, { ... })
+---Run action, this function is asynchronous
+---@param name string The action name, generally a vscode command
+---@param opts? table Optional options table, all fields are optional
+---            - args: (table) Optional arguments for the action
+---            - range: (table) Specific range for the action. In visual mode, this parameter is generally not required.
+---                     The format is consistent with the LSP Protocol.
+---            - line_range: (table) An array [start_line, end_line], which has the same effect as range. Lines are zero indexed
+---            - leave_selection: (boolean) Whether to preserve the selected range, only valid when range or line_range is specified
+---            - callback: (function(err: string|nil, ret: any))
+---                        Optional callback function to handle the action result.
+---                        The first argument is the error message, and the second is the result.
+---                        If no callback is provided, any error message will be shown as a notification in VSCode.
+function M.action(name, opts)
+  vim.validate({
+    name = { name, "string" },
+    opts = { opts, "table", true },
+  })
+  opts = opts or {}
+  vim.validate({
+    ["opts.callback"] = { opts.callback, "f", true },
+    ["opts.args"] = { opts.args, "t", true },
+    ["opts.range"] = { opts.range, "t", true },
+    ["opts.line_range"] = { opts.lien_range, "t", true },
```

Comment:
line_range, not lien_range

Reply:
If you use simple keibindings, you won't use the callback parameter.